provider/clusterrosaclassic/cluster_rosa_classic_resource.go

```diff
@@ -1666,15 +1668,29 @@ func populateRosaClassicClusterState(ctx context.Context, object *cmv1.Cluster,
 			state.Proxy.NoProxy = types.StringValue(noProxy)
 		}
 	} else {
-		state.Proxy = nil
+		// We cannot set the proxy to nil because the attribute state.Proxy.AdditionalTrustBundle might contain a value.
+		// Due to the sensitivity of this attribute, the backend returns the value REDIRECT for a non-empty AdditionalTrustBundle
+		// and if state.Proxy is null it will override the actual value.
+		hasProxy = false
 	}
 
 	trustBundle, ok := object.GetAdditionalTrustBundle()
-	if ok && common.IsStringAttributeUnknownOrEmpty(state.Proxy.AdditionalTrustBundle) {
-		if state.Proxy == nil {
-			state.Proxy = &proxy.Proxy{}
+	if ok {
+		// update the state only if the current state is Null or Empty
+		// it can happen after `import` command or when it was updated from a different cli tool
+		if state.Proxy == nil || common.IsStringAttributeKnownAndEmpty(state.Proxy.AdditionalTrustBundle) {
```

Comment:
I don't think this if statement is required
Is there any reason why not to set AdditionalTrustBundle when it already exist in the state?